File: app/src/main/java/org/fossasia/susi/ai/rest/services/DeviceService.java
package org.fossasia.susi.ai.rest.services;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;


public interface DeviceService {

    @GET("/wifi_credentials/{wifissid}/{wifipassid}")
    Call<String> wifiCredentials(@Path("wifissid") String ssid,
                                     @Path("wifipassd") String pass);


    @GET("/config/{stt}/{tts}/{hotword}/{wake}")
    Call<String> ttSSettings(@Path("stt") String speechToText,
                                @Path("tts") String textToSpeech,
                                @Path("hotword") String hotword,
                                @Path("wake") String wake);


    @GET("/auth/{choice}/{email}/{password}")
    Call<String> authCredentials(@Path("choice") String choice,
                                     @Path("email") String email,
                                     @Path("password") String password);

}

File: app/src/main/java/org/fossasia/susi/ai/data/DeviceService.kt
package org.fossasia.susi.ai.data

import org.fossasia.susi.ai.data.contract.IDeviceModel
import org.fossasia.susi.ai.dataclasses.SpeakerConfiguration

interface DeviceService {

    fun wifiCredentials(ssid: String, pass: String, listener: IDeviceModel.onSendWifiCredentialsListener)

    fun ttsSettings(speakerConfig: SpeakerConfiguration, listener: IDeviceModel.onSetConfigurationListener)

    fun authCredentials(choice: String, email: String, password: String, listener: IDeviceModel.onSendAuthCredentialsListener)
}